src/VirtualClient/VirtualClient.Actions/MLPerfTraining/MLPerfTrainingExecutor.cs

```diff
@@ -0,0 +1,511 @@
+        }
+
+        /// <summary>
+        /// The MLPerf Training Pytorch code directory.
+        /// </summary>
+        protected string ExecutionDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.NvidiaDirectoryPath, this.Model, "implementations", this.Implementation);
+            }
+        }
+
+        /// <summary>
+        /// The MLPerf Training Nvidia data directory.
+        /// </summary>
+        protected string DataDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.PlatformSpecifics.PackagesDirectory, this.Model);
+            }
+        }
+
+        /// <summary>
+        /// The output directory of MLPerf.
+        /// </summary>
+        protected string OutputDirectoryPath
+        {
+            get
+            {
+                return this.PlatformSpecifics.Combine(this.ExecutionDirectoryPath, "results");
+            }
+        }
+
+        /// <summary>
+        /// Initializes the environment for execution of the MLPerf Training workload.
+        /// </summary>
+        protected override async Task InitializeAsync(EventContext telemetryContext, CancellationToken cancellationToken)
+        {
+            this.Logger.LogTraceMessage($"{this.TypeName}.InitializationStarted", telemetryContext);
+
+            this.ThrowIfPlatformNotSupported();
+
+            await this.ThrowIfUnixDistroNotSupportedAsync(cancellationToken)
+                .ConfigureAwait(false);
+
+            MLPerfState state = await this.stateManager.GetStateAsync<MLPerfState>($"{nameof(MLPerfState)}", cancellationToken)
+                ?? new MLPerfState();
+
+            if (!state.Initialized)
+            {
+                // add user in docker group
+                await this.ExecuteCommandAsync("usermod", $"-aG docker {this.Username}", this.ExecutionDirectoryPath, cancellationToken);
+                await this.ExecuteCommandAsync("newgrp", $"docker", this.ExecutionDirectoryPath, cancellationToken);
+
+                // If GPUConfig is not included in the MLPerf code but is supported
+                this.ReplaceGPUConfigFilesToSupportAdditionalGPUs();
+
+                // Setup Environment
+                await this.SetupEnvironmentAsync(cancellationToken);
+                state.Initialized = true;
+                await this.stateManager.SaveStateAsync<MLPerfState>($"{nameof(MLPerfState)}", state, cancellationToken);
+            }
+        }
+
+        /// <summary>
+        /// Creates setup for MLPerf Training workload.
+        /// </summary>
+        /// <param name="cancellationToken"></param>
+        /// <returns></returns>
+        protected async Task SetupEnvironmentAsync(CancellationToken cancellationToken)
+        {
+            string dockerImageCommand = $"docker build --pull -t {this.GetContainerName()} .";
+            string configCommand = $"bash ./config_DGXA100_1x8x56x1.sh";
```

Comment:
This script will change for every HW right? By name it suggests that this is only applicable for DGXA100 HW with 8 GPUs, should we keep this name according to on which HW we are running the workload? If it is not possible to detect as of now, we can atleast keep it as parameter so that we can control it in experiments while launching via Juno/Vega

Reply:
Yes, I have put that as a parameter in the profile